### PR TITLE
#557 [Ruby 2.5] Update specs to use frozen_error_class for FrozenError

### LINKED ISSUE #557: [Ruby 2.5] FrozenError
A huge bunch on specs should be modified to spec Ruby 2.5's behavior of raising a `FrozenError` when attempting to mutate a frozen object.

Instead duplicating the affected specs and referring to `FrozenError` directly, the helper `frozen_error_class` should be used (see [this](https://github.com/ruby/mspec/pull/35)). As such, no direct mention of `FrozenError` should appear in the specs, only `frozen_error_class`.

Anyone wants to handle this?

### PR DESCRIPTION
**Pull request by Atul9**
closes #557

### CODE REVIEW

File: optional/capi/object_spec.rb [hunk 1]
```diff
@@ -666,14 +666,14 @@ def reach
       obj.tainted?.should == true
     end
 
-    it "raises a RuntimeError if the object passed is frozen" do
-      lambda { @o.rb_obj_taint("".freeze) }.should raise_error(RuntimeError)
+    it "raises a FrozenError if the object passed is frozen" do
```
Comment (eregon) [inline optional/capi/object_spec.rb]: Should be `#{frozen_error_class}`

File: optional/capi/object_spec.rb [hunk 2]
```diff
@@ -666,14 +666,14 @@ def reach
...
+      lambda { @o.rb_obj_taint("".freeze) }.should raise_error(frozen_error_class)
     end
   end
 
   describe "rb_check_frozen" do
-    it "raises a RuntimeError if the obj is frozen" do
-      lambda { @o.rb_check_frozen("".freeze) }.should raise_error(RuntimeError)
+    it "raises a frozen_error_class if the obj is frozen" do
```
Comment (eregon) [inline optional/capi/object_spec.rb]: Should be `#{frozen_error_class}`

File: optional/capi/array_spec.rb [hunk 3]
```diff
@@ -214,9 +214,9 @@
       a.should == [nil, nil, 7]
     end
 
-    it "raises a RuntimeError if the array is frozen" do
+    it "raises a FrozenError if the array is frozen" do
```
Comment (eregon) [inline optional/capi/array_spec.rb]: Should be `#{frozen_error_class}`

Comment (Atul9) [reply optional/capi/array_spec.rb]: done. I have updated the PR.

File: optional/capi/object_spec.rb (hunk 1 above)
Comment (Atul9) [reply optional/capi/object_spec.rb]: done. I have updated the PR.

File: optional/capi/object_spec.rb (hunk 2 above)
Comment (Atul9) [reply optional/capi/object_spec.rb]: done. I have updated the PR.

**Pull request merged**